frontend/src/app/account/profile/profile.component.ts

```diff
@@ -0,0 +1,241 @@
+import { Router } from '@angular/router';
+
+@Component({
+    standalone: true,
+    imports: [
+        CommonModule,
+        ReactiveFormsModule,
+        ButtonModule,
+        PasswordModule,
+        ToastModule,
+        DividerModule,
+        InputTextModule,
+    ],
+    providers: [MessageService],
+    templateUrl: './profile.component.html',
+    styleUrls: ['./profile.component.css'],
+})
+export class ProfileComponent implements OnInit {
+    user: User | null = null;
+    showEditProfile = false;
+    showEditPassword = false;
+    isProcessingEdit = false;
+    isProcessingPassword = false;
+
+    constructor(
+        private authenticationService: AuthenticationService,
+        private messageService: MessageService,
+        private router: Router,
+        public formUtils: FormUtilsService,
+    ) {}
+
+    editProfileForm: FormGroup = new FormGroup({
+        username: new FormControl('', [Validators.required, invalidUsernameValidator()]),
+        email: new FormControl('', [Validators.required, Validators.email]),
+        password: new FormControl(''),
+    });
+
+    editPasswordForm: FormGroup = new FormGroup(
+        {
+            oldPassword: new FormControl(''),
+            password: new FormControl('', [
+                Validators.required,
+                invalidPasswordValidator(),
+                lowercasePasswordValidator(),
+                uppercasePasswordValidator(),
+                numericPasswordValidator(),
+                specialPasswordValidator(),
+                shortPasswordValidator(),
+                weakPasswordValidator(),
+            ]),
+            confirmPassword: new FormControl('', [Validators.required]),
+        },
+        {
+            validators: mismatchPasswordValidator('password', 'confirmPassword'),
+        },
+    );
+
+    passwordRequirements = [
+        { msg: 'At least one lowercase', check: () => this.formUtils.passwordHasNoLowercase(this.editPasswordForm) },
+        { msg: 'At least one uppercase', check: () => this.formUtils.passwordHasNoUppercase(this.editPasswordForm) },
+        { msg: 'At least one numeric', check: () => this.formUtils.passwordHasNoNumeric(this.editPasswordForm) },
+        {
+            msg: 'At least one special character',
+            check: () => this.formUtils.passwordHasNoSpecial(this.editPasswordForm),
+        },
+        { msg: 'Minimum 8 characters', check: () => this.formUtils.isPasswordShort(this.editPasswordForm) },
+    ];
+
+    ngOnInit() {
+        this.user = this.authenticationService.userValue;
+        this.showEditProfile = false;
+        this.showEditPassword = false;
+        this.isProcessingEdit = false;
+        this.isProcessingPassword = false;
+    }
+
+    onUpdateProfile() {
+        this.showEditProfile = true;
+        this.showEditPassword = false;
+    }
+
+    onUpdatePassword() {
+        this.showEditPassword = true;
+        this.showEditProfile = false;
+    }
+
+    onEditSubmit() {
```

Comment:
I think the users are in need of a success message for both kinds of edits

Reply:
I call a login() to update after successful edit of profile or password but without refreshing the page, it doesnt update properly. I tried to give success messages, but they do not persist when I refresh the page.

Reply:
This is TODO ig. I have to think of a way to do it better to show success message.

Reply:
Sorry not too sure what you mean here. It seems to be working for me?

https://github.com/user-attachments/assets/8f9a18cd-b8d9-4d68-9619-069a58e11071

Reply:
yep mb